src/net_processing.cpp

```diff
@@ -1551,48 +1642,8 @@ bool static ProcessHeadersMessage(CNode *pfrom, CConnman *connman, const std::ve
     CValidationState state;
     CBlockHeader first_invalid_header;
     if (!ProcessNewBlockHeaders(headers, state, chainparams, &pindexLast, &first_invalid_header)) {
```

Comment:
nit: drop the `first_invalid_header` argument. It defaults to `nullptr` and your no longer using the returned value.

Follow-up: remove the `first_invalid` argument from `ProcessNewBlockHeaders()` since none of the callers now use it.